modular_skyrat/modules/pixel_shift/code/pixel_shift_mob.dm

```diff
@@ -0,0 +1,24 @@
+/mob/Initialize(mapload)
+	. = ..()
+	add_pixel_shift_component()
+
+/mob/proc/add_pixel_shift_component()
+	return
+
+/mob/living/add_pixel_shift_component()
+	AddComponent(/datum/component/pixel_shift)
```

Comment:
you're kinda crazy adding a component to every living mob in init rather than only adding it when someone presses the shift hotkey

Reply:
How about I move it to Login, so it guarantees that a client is there and there won't be a need to check it everytime hotkey is pressed?

Reply:
Login is better but you really could just put it in `/datum/keybinding/mob/pixel_shift/down` and delete it in `/datum/keybinding/mob/pixel_shift/up`, I don't see any obvious need to keep the component on mobs which aren't shifting (unless I'm missing something?)

Reply:
> 

Wouldn't it be heavy to add/remove component if the button is spammed?

Reply:
And I need to constantly check if user input wants to pixel_shift or to move and unpixel_shift.

I need to keep the component and signals for "activate shifting" and "disable shifting".
Component:
1. Checks if the user is shifted when Moved - then does unpixel_shift()
2. Checks if the user is still holding Pixel Shift keybind - do pixel_shift() if they do OR do move() if the don't and unpixel_shift()

I need to somehow keep track of 2, so qdel-ing on up() doesn't seem good unless I miss something obvious

Reply:
> Wouldn't it be heavy to add/remove component if the button is spammed?

Not really. We do worse in other places

> I need to somehow keep track of 2, so qdel-ing on up() doesn't seem good unless I miss something obvious

Good point, you'd only want to delete the component (`qdel(src)`) when you stop shifting (in `unpixel_shift()`)